### PR TITLE
Small improvements on `script/rspec-slow-repeat` file

### PR DESCRIPTION
**Pull request by jibees**
#### What? Why?

 - `/proc/cpuinfo` is not available on Darwin arch (ie. mac) 

#### What should we test?
 - run `script/rspec-slow-repeat` on both linux and mac. Should work. 

#### Release notes

Changelog Category: Technical changes

### CODE REVIEW

**Comment by jibees** [comment]
@mkllnk 
I'm wondering why/what this code is supposed to do:
```bash
for i in `seq $processors`; do
  yes > /dev/null &
done
```

**Comment by mkllnk** [comment]
The `yes` command just outputs "yes" as many times as it can. The purpose here is to occupy the CPU. But it's not multi-threaded and it occupies only one core. So we start one process for each core and redirect the output to /dev/null which is discarding the output. The result is a 100% busy CPU, simulating a very busy CI environment, increasing the chance of race conditions when executing specs.

File: script/rspec-slow-repeat [hunk 1]
```diff
@@ -15,7 +15,17 @@ fi
 
 n="$1"
 passed=0
-processors="`cat /proc/cpuinfo | grep -c processor`"
+
+# Check via uname the environment we are running in to get the number of cores
+if [[ "`uname`" == "Darwin" ]]; then
+  processors="`sysctl -n hw.ncpu.`"
+  env="mac"
```
Comment (mkllnk) [inline script/rspec-slow-repeat:22]: We don't really need this variable, do we?

**Review by mkllnk — APPROVED** [review]
Shall we add an explanation to the "yes" part?

**Comment by jibees** [comment]
Thank you for your explanation!

Capture d’écran 2022-10-31 à 10 09 44

Even if I have 12 cores, and the script launch 12 `yes` command, only one of two cores are 100% busy. 
But nevermind, I don't think it's really important.

File: script/rspec-slow-repeat (hunk 1 above)
Comment (jibees) [reply script/rspec-slow-repeat:22]: No that's not that important. That was mainly for debug.

**Review by filipefurtad0 — APPROVED** [review]
Thanks @jibees :pray:

**Review by mkllnk — APPROVED** [review]
Wow, heaps better, thank you!

Occupying your CPU doesn't seem to work that well. Seems like Mac is too clever and using resources too efficiently! :wink:

**Pull request merged**